specification/loadtestservice/LoadTestService/models.tsp

```diff
@@ -283,40 +347,127 @@ not required to upload.
 
   @doc("Optional load test config")
   optionalLoadTestConfig?: OptionalLoadTestConfig;
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Type of test. e.g. QuickTest, UrlBasedTest")
+  testType?: TestType;
+}
+
+@doc("Quick test config")
+model QuickTestConfig {
+  @doc("Test URL. Provide the complete HTTP URL. For example, http://contoso-app.azurewebsites.net/login")
+  endpointUrl?: string;
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Throughput required (requests per second)")
+  rps?: int32;        
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("Maximum response time of the API/endpoint")
+  maxResponseTime?: int32;
+}
+
+@doc("Url test config")
+model UrlTestConfig {
```

Comment:
nit: This doesnt look expandable or generic. We can make these as LoadParameteres with Key and Value ad/or enums.

Reply:
Also can we use some sort of polymorphism.. somethign similar to below ?
```
model UrlTestConfig  {
  loadType: LoadType
  loadTypeConfiguration: DefaultLoadTypeConfiguration?
}

DefaultLoadTypeConfiguration
{
}

StepUpLoadTypeConfigruation
{
  @added(APIVersions.v2023_04_01_preview)
  @doc("Number of steps in the ramp - up phase")
  rampUpSteps?: int32;  
}

SpikeLoadTypeConfiguration : DefaultLoadTypeConfiguration
{
spikeHoldLoad: int32
spikeMultiplier: int32
}

```

Reply:
Thanks for the suggestions, we are already exploring the option for Polymorphism for the models OptionalLoadConfig, URLBasedConfig and QuickTestConfig. The discussions are ongoing with API Spec review team.